src/main/java/winstone/Launcher.java

```diff
@@ -229,9 +231,19 @@ private void writePortToFileIfNeeded() throws IOException {
                     Path portFile = Paths.get(portFileName);
                     Path portDir = portFile.getParent();
                     Files.createDirectories(portDir);
-                    try (BufferedWriter writer = Files.newBufferedWriter(portFile, StandardCharsets.UTF_8)) {
+                    Path tmpPath = Files.createTempFile(portDir, "temp", "");
+                    try (BufferedWriter writer = Files.newBufferedWriter(tmpPath, StandardCharsets.UTF_8)) {
                         writer.write(Integer.toString(port));
                     }
+
+                    try {
+                        Files.move(tmpPath, portFile, StandardCopyOption.ATOMIC_MOVE);
+                    } catch (AtomicMoveNotSupportedException e) {
+                        Logger.logDirectMessage(Level.WARNING, "", "Atomic move does not supported", e);
+                        Files.move(tmpPath, portFile, StandardCopyOption.REPLACE_EXISTING);
+                    } finally {
+                        tmpPath.toFile().deleteOnExit();
```

Comment:
The deletion here will always be a no-op, as the temporary path no longer exists at this point. So there is no reason to try and delete it. Suggest simply removing this unnecessary code:

```suggestion
```

Reply:
remove redundant deleteOnExit